app/controllers/requests_controller.rb

```diff
@@ -36,24 +36,19 @@ def create
   end
 
   def new
-    @request = Request.new(organization: @organization)
+    @request = @organization.requests.new
   end
 
   def edit; end
 
   def update
     @request.files.purge_later if @request.files.attached? && request_params[:files].blank?
     if @request.update(request_params)
-      if @request.planned?
-        redirect_to organization_requests_path(@request.organization_id, filter: :planned), flash: {
-          success: I18n.t('request.schedule_request_success',
-                          count: Contributor.active.with_tags(@request.tag_list).count,
-                          scheduled_datetime: I18n.l(@request.schedule_send_for, format: :long))
-        }
-      else
-        redirect_to organization_request_path(@request.organization_id, @request),
-                    flash: { success: I18n.t('request.success', count: @request.stats[:counts][:recipients]) }
-      end
+      redirect_to organization_requests_path(@request.organization_id, filter: :planned), flash: {
+        success: I18n.t('request.schedule_request_success',
+                        count: @request.organization.contributors.active.with_tags(@request.tag_list).count,
+                        scheduled_datetime: I18n.l(@request.schedule_send_for, format: :long))
+      }
```

Comment:
I removed the conditional here because the first thing we do before hitting the update method is ensure that the request is planned for the future. We disallow editing for requests that have already been broadcast.

Reply:
Is it possible to update a former planned request to be sent now? In that case that might still have to stay...

Reply:
Good catch! The only thing that is different is the success_message and the filter, but it is not good UX to be redirected to the incorrect path so I've added back complexity to improve the UX.

I've also added a test to not forget the purpose of the complexity in the future. See 7131b583